src/Engine/ProtoCore/Utils/ArrayUtils.cs

```diff
@@ -117,18 +117,25 @@ public static ClassNode GetGreatestCommonSubclassForArray(StackValue array, Runt
             return runtimeCore.DSExecutable.classTable.ClassNodes[orderedTypes.First()];
         }
 
-        public static Dictionary<int, StackValue> GetTypeExamplesForLayer(StackValue array, RuntimeCore runtimeCore)
+        /// <summary>
+        /// This method returns the distinct reduced params for all the elements inside the
+        /// currentReducedParam, if it is an array. If it is not an array, it just returns the currentReducedParam.
+        /// </summary>
+        /// <param name="currentReducedParam"></param>
+        /// <param name="runtimeCore"></param>
+        /// <returns> A dictionary where the value is the current ReducedParam and the key is its metaData type</returns>
+        public static Dictionary<int, StackValue> GetTypeExamplesForLayer(StackValue currentReducedParam, RuntimeCore runtimeCore)
```

Comment:
I would avoid calling these reduced params. Even though this method is called while computing them, the method itself is independent from that.